transaction/std/src/transaction_builder.rs

```diff
@@ -126,11 +131,18 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
             .memo_builder
             .take()
             .expect("memo builder is missing, this is a logic error");
+        let block_version = self.block_version.clone();
         let result = self.add_output_with_fog_hint_address(
             value,
             recipient,
             recipient,
-            |memo_ctxt| mb.make_memo_for_output(value, recipient, memo_ctxt),
+            |memo_ctxt| {
+                if block_version.e_memo_feature_is_supported() {
```

Comment:
This is where the transaction builder uses block version to decide whether or not to filter out memos